igvc_navigation/src/mapper/unrolling_layer.cpp

```diff
@@ -0,0 +1,249 @@
+  const auto origin_y = static_cast<double>(current_metadata_->origin.position.y);
+
+  const auto update_origin_x = origin_x + (map->x * resolution);
+  const auto update_origin_y = origin_y + (map->y * resolution);
+
+  // Calculate where the update map's origin is on our map
+  const int map_idx_x = static_cast<int>((update_origin_x - getOriginX()) / resolution);
+  const int map_idx_y = static_cast<int>((update_origin_y - getOriginY()) / resolution);
+
+  size_t start_idx_x = 0;
+  size_t start_idx_y = 0;
+  // If map_idx is negative, then calculate the offset we need to start iterating map at
+  // so that we don't go off the map
+  if (map_idx_x < 0)
+  {
+    start_idx_x = -map_idx_x;
+  }
+  if (map_idx_y < 0)
+  {
+    start_idx_y = -map_idx_y;
+  }
+
+  // Calculate where map's end is on our map
+  const size_t our_idx_end_x = map_idx_x + length_x;
+  const size_t our_idx_end_y = map_idx_y + length_y;
+
+  const auto last_idx_x = getSizeInCellsX();
+  const auto last_idx_y = getSizeInCellsY();
+
+  size_t end_idx_x = length_x;
+  size_t end_idx_y = length_y;
+
+  // If map's end is larger than our largest, then last index to iterate is our largest - their origin in our map
+  if (our_idx_end_x > last_idx_x)
+  {
+    end_idx_x = last_idx_x - map_idx_x;
+  }
+  if (our_idx_end_y > last_idx_y)
+  {
+    end_idx_y = last_idx_y - map_idx_y;
+  }
+
+  UpdateMapMetadata metadata{
+    map_idx_x, map_idx_y, start_idx_x, start_idx_y, end_idx_x, end_idx_y, length_x, length_y
+  };
+  updateMap(map->data, metadata);
+}
+
+void UnrollingLayer::updateMap(const std::vector<int8_t>& map, const UpdateMapMetadata& metadata)
+{
+  touch(metadata.map_idx_x + metadata.start_idx_x, metadata.map_idx_y + metadata.start_idx_y,
+        metadata.map_idx_x + metadata.end_idx_x, metadata.map_idx_y + metadata.end_idx_y);
+  for (size_t y = metadata.start_idx_y; y < metadata.end_idx_y; y++)
+  {
+    const size_t our_y = metadata.map_idx_y + y;
+    const size_t map_index_base = y * metadata.length_x;
+    const size_t our_index_base = our_y * getSizeInCellsX();
+    for (size_t x = metadata.start_idx_x; x < metadata.end_idx_x; x++)
+    {
+      const size_t our_x = metadata.map_idx_x + x;
+      const size_t map_index = map_index_base + x;
+      const size_t our_index = our_index_base + our_x;
+
+      costmap_[our_index] = translator[static_cast<uint8_t>(map[map_index])];
+    }
+  }
+}
+
+void UnrollingLayer::touch(size_t start_x, size_t start_y, size_t end_x, size_t end_y)
+{
+  min_map_x_ = std::min(min_map_x_, start_x);
+  min_map_y_ = std::min(min_map_y_, start_y);
+  max_map_x_ = std::max(max_map_x_, end_x);
+  max_map_y_ = std::max(max_map_y_, end_y);
+}
+
+void UnrollingLayer::updateBounds(double robot_x, double robot_y, double robot_yaw, double* min_x, double* min_y,
+                                  double* max_x, double* max_y)
+{
+  ROS_INFO_STREAM(">>> updateBounds");
```

Comment:
Could you remove these info prints? It doesn't provide much information now that you're done debugging and they get very spammy in the info stream.

Reply:
oh woops I forgot that I added these in, mb